_posts/2020-03-23-one-click-install.md

```diff
@@ -0,0 +1,34 @@
+---
+layout: blog-detail
+post-type: blog
+by: Sébastien Doeraene
+title: "One-click install for Scala"
+---
+
+Installing Scala has always been a task more challenging than necessary, with the potential to drive away beginners.
+Should I install Scala itself? sbt? Some other build tools? What about a better REPL like Ammonite? Oh and before all that I need to install Java?
+
+To solve this problem, the Scala Center contracted Alexandre Archambault in January 2020 to add a one-click install of Scala to [coursier](https://get-coursier.io/).
+Starting today, [the main Scala download page](/download/) gives one-liner command lines incantations (and a one-click executable for Windows) to fully install Scala and all the main related tools.
+For example, on Linux, all we now need is:
+
+```bash
+$ curl -Lo cs https://git.io/coursier-cli-linux && chmod +x cs && ./cs setup
+```
+
+which will install all the following software, if not yet installed:
+
+* a JDK
+* the build tools [sbt](https://www.scala-sbt.org/) and [mill](https://www.lihaoyi.com/mill/)
+* the [Ammonite](https://ammonite.io/) enhanced REPL
+* [scalafmt](https://scalameta.org/scalafmt/), the Scala formatter
+* the [coursier](https://get-coursier.io/) CLI, to install further Scala-based applications
+* the `scala` and `scalac` command-line tools
```

Comment:
Should we add some details about how we manage specific versions of `scala` and `scalac`, and how to update the `cs` tool itself?

Reply:
Done as well.